crates/libs/link/src/lib.rs

```diff
@@ -0,0 +1,39 @@
+#![doc = include_str!("../readme.md")]
+#![no_std]
+
+/// Defines an external function to import.
+#[cfg(all(windows, target_arch = "x86"))]
+#[macro_export]
+macro_rules! link {
+    ($library:literal $abi:literal $($link_name:literal)? fn $($function:tt)*) => (
+        #[link(name = $library, kind = "raw-dylib", modifiers = "+verbatim", import_name_type = "undecorated")]
+        extern $abi {
+            $(#[link_name=$link_name])?
+            pub fn $($function)*;
+        }
+    )
+}
+
+/// Defines an external function to import.
+#[cfg(all(windows, not(target_arch = "x86")))]
+#[macro_export]
+macro_rules! link {
+    ($library:literal $abi:literal $($link_name:literal)? fn $($function:tt)*) => (
+        #[link(name = $library, kind = "raw-dylib", modifiers = "+verbatim")]
+        extern "C" {
```

Comment:
This is just a comment, I had to look this up to recall why `extern "C"` is used instead of `$abi`. It was https://github.com/rust-lang/rust/issues/110505 which wasn't fixed until 1.77, a fair bit above the current MSRV.

Reply:
No, that cannot be true. https://github.com/rust-lang/rust/issues/110505 got closed by https://github.com/rust-lang/rust/pull/119587 which just adds an unstable feature that's still unstable.